ext/tunemygc/tunemygc_ext.c

```diff
@@ -40,6 +61,33 @@ static double _tunemygc_walltime()
   return ts.tv_sec + ts.tv_nsec * 1e-9;
 }
 
+#ifdef RUBY_INTERNAL_EVENT_GC_ENTER
+static struct {
+    double last_start;
+    unsigned long gc_count;
+    double gc_time;
+    double last_reset;
+} counters;
+
+static void reset_counters() {
+    counters.last_reset = _tunemygc_realtime();
+    counters.gc_time = 0;
+    counters.gc_count = 0;
+}
+
+static VALUE tunemygc_counters(VALUE mod)
+{
+    VALUE rb_counters = rb_ary_new2(3);
+    rb_ary_store(rb_counters, 0, ULONG2NUM((unsigned long) (counters.gc_time * 1000)));
+    rb_ary_store(rb_counters, 1, DBL2NUM(counters.last_reset));
+    rb_ary_store(rb_counters, 2, ULONG2NUM(counters.gc_count));
```

Comment:
I _think_, on x86, we're good an don't need a mfence while reading these values of `counters`